vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -251,6 +266,102 @@ def _update_states(self, scheduler_output: SchedulerOutput):
                 req_index,
                 start_token_index:end_token_index] = req_data.new_token_ids
 
+        if scheduler_output.finished_req_ids:
+            for req_id in scheduler_output.finished_req_ids:
+                self.input_batch.remove_request(req_id)
+                self.requests.pop(req_id, None)
+            self.input_batch.refresh_sampling_metadata()
+
+    def prepare_prompt(
+        self,
+        new_requests: list[NewRequestData],
+    ) -> ModelForwardInputs:
+        raise NotImplementedError
+
+    def prepare_decode(
+        self,
+        cached_requests: list[CachedRequestData],
+    ) -> ModelForwardInputs:
+        raise NotImplementedError
+
+    def prepare_model_input(
+            self, scheduler_output: SchedulerOutput) -> ModelForwardInputs:
+
+        # NOTE: We assume that all sequences in the group are all prompts or
+        # all decodes. Also assuming that new sequences are prefills
+        is_prompt = len(scheduler_output.scheduled_new_reqs) > 0
+
+        # Prepare input tensors.
+        if is_prompt:
+            # Assert no running requests
+            assert len(scheduler_output.scheduled_cached_reqs) == 0
+
+            return self.prepare_prompt(scheduler_output.scheduled_new_reqs)
+        else:
+            return self.prepare_decode(scheduler_output.scheduled_cached_reqs)
+
+    def base_execute_model(self, scheduler_output: SchedulerOutput,
+                           **kwargs) -> ModelRunnerOutput:
+
+        t0 = time.time()
+
+        self.update_states(scheduler_output)
+
+        if not scheduler_output.total_num_scheduled_tokens:
+            # Return empty ModelRunnerOuptut if there's no work to do.
+            return EMPTY_MODEL_RUNNER_OUTPUT
+
+        model_input = self.prepare_model_input(scheduler_output)
+
+        # Execute the model
+        attn_metadata = self.build_attn_metadata(model_input)
```

Comment:
Uhh, maybe I didn't understand it, but in the function `build_attn_metadata`, we don't really do anything with the `model_input`. So why are we passing it here?